qiskit/circuit/quantumcircuit.py

```diff
@@ -1775,6 +1783,58 @@ def barrier(self, *qargs):
 
         return self.append(Barrier(len(qubits)), qubits, [])
 
+    def delay(self, duration, qarg=None, unit='dt'):
+        """Apply :class:`~qiskit.circuit.Delay`. If qarg is None, applies to all qubits.
+        When applying to multiple qubits, delays with the same duration will be created.
+
+        Args:
+            duration (int or float or ParameterExpression): duration. Integer type indicates
+                duration is unitless, i.e. use dt of backend.
+                In the case of float, its `unit` must be specified.
+            qarg (Object): qubit argument to apply this delay.
+            unit (str): unit of the duration. Default unit is ``dt``, which depends on backend.
+
+        Returns:
+            qiskit.Instruction: the attached delay instruction.
+
+        Raises:
+            CircuitError: if arguments have bad format.
+        """
+        from .delay import Delay
+        qubits = []
+        if qarg is None:  # -> apply delays to all qubits
+            for qreg in self.qregs:
+                for j in range(qreg.size):
+                    qubits.append(qreg[j])
+        else:
+            if isinstance(qarg, QuantumRegister):
+                qubits.extend([qarg[j] for j in range(qarg.size)])
+            elif isinstance(qarg, list):
+                qubits.extend(qarg)
+            elif isinstance(qarg, (range, tuple)):
+                qubits.extend(list(qarg))
+            elif isinstance(qarg, slice):
+                qubits.extend(self.qubits[qarg])
+            else:
+                qubits.append(qarg)
+
+        if isinstance(duration, float):
+            if unit == 'dt':
+                raise CircuitError('duration in dt must be integer.')
+        else:
+            if not isinstance(duration, (int, ParameterExpression)):
+                raise CircuitError('Invalid duration type.')
+
+        if unit not in {'dt', 's', 'ms', 'us', 'µs', 'ns', 'ps'}:
+            raise CircuitError('Unknown unit is specified.')
+
+        instructions = InstructionSet()
+        for q in qubits:
+            added = self.append(Delay(duration, unit), [q])
+            for inst in added:
```

Comment:
This would be much easier to read.
```
instructions = InstructionSet()
for q in qubits:
    inst = (Delay(duration, unit), [q], [])
    self.append(*inst)
    instructions.add(*inst)
```
(Anyway, I'm not a big fan of InstructionSet...)